lib/oop.js

```diff
@@ -23,7 +23,36 @@ const mixin = (target, source) => {
   Object.assign(target, mix);
 };
 
+const ASCII_A = 65;
+
+// Convert instance with public fields to instance with private fields
+//   instance - <Object>, source instance
+// Returns: <Object> - destination instance
+//
+// Example: common.privatize({ private: 5, f() { return this.private; } });
+const privatize = instance => {
+  const iface = {};
+  const fields = Object.keys(instance);
+  for (const fieldName of fields) {
+    const field = instance[fieldName];
+    if (typeof field === 'function') {
+      const bindedMethod = field.bind(instance);
+      iface[fieldName] = bindedMethod;
+    } else if (fieldName === fieldName.toUpperCase()) {
+      const first = fieldName.charCodeAt(0);
+      if (first >= ASCII_A) {
+        Object.defineProperty(iface, fieldName, {
+          enumerable: true,
+          get: () => field,
+        });
+      }
+    }
+  }
+  return Object.freeze(iface);
+};
```

Comment:
what about private methods?
```js
const dest = privatize({
  _privateMethod() {},
  publicMethod() { this._privateMethod(); },
});
```
Many developers could expect that the dest contains only `publicMethod`.
So, I suggest to extract logic of private fields recogintion to the dedicated function and pass it to the `privatize` as a parameter.

```js
const ASCII_A = 'A'.charCodeAt(0);
const ASCII_Z = 'Z'.charCodeAt(0);

const isExplicitlyPrivate = name => name.charAt(0) === '_';

const isMethod = value => typeof value === 'function';

const isConstantName = name => {
  if (name !== name.toUpperCase()) return false;
  const firstCharCode = name.charCodeAt(0);
  return firstCharCode >= ASCII_A && firstCharCode <= ASCII_Z;
};

const isPublicField = (name, value) => isMethod(value) || isConstantName(name);

const isPrivateField = (name, value) =>
  isExplicitlyPrivate(name) || !isPublicField(name, value);

const privatize = (instance, isPrivate = isPrivateField) => {
  const iface = {};
  const fields = Object.keys(instance);
  for (const fieldName of fields) {
    const fieldValue = instance[fieldName];

    if (isPrivate(fieldName, fieldValue)) continue; // skipping private fields

    iface[fieldName] = isMethod(fieldValue)
      ? fieldValue.bind(instance)
      : fieldValue;
  }
  return Object.freeze(iface);
};
```